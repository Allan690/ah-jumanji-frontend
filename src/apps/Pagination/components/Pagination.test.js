import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Pagination from "./Pagination";
import PropTypes from "prop-types";

Enzyme.configure({adapter: new Adapter()});
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, {context});
};

const props = {
    totalRecords: 20,
    pageLimit: 5,
    pageNeighbours: 1,
    onPageChanged: jest.fn()
  };

describe('<Pagination />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<Pagination {...props} debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('<Pagination />', () => {
  let store;
  let wrapper;

  const props = {
    totalRecords: 20,
    pageLimit: 5,
    pageNeighbours: 1,
    onPageChanged: jest.fn()
  };
  const initialState = {};

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Pagination {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('renders without crashing', () => {
    shallowWithStore(<Pagination {...props} />, store);
  });

  it('should render the component correctly', () => {
    expect(wrapper.find('.pagination').exists()).toBe(true);
    expect(wrapper.find('button').exists()).toBe(true);
  });

});
