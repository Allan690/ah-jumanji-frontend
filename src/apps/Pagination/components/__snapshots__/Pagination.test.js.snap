// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match the snapshot 1`] = `
<Fragment>
  <nav
    aria-label="Articles Pagination"
  >
    <ul
      className="pagination pagination-sm"
      id="paginate"
    >
      <li
        className="page-item active"
        key="0"
      >
        <button
          className="page-link"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </li>
      <li
        className="page-item"
        key="1"
      >
        <button
          className="page-link"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </li>
      <li
        className="page-item"
        key="2"
      >
        <button
          className="page-link"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </li>
      <li
        className="page-item"
        key="3"
      >
        <button
          className="page-link"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </li>
    </ul>
  </nav>
</Fragment>
`;
